iac-common/src/main/java/org/sonar/iac/common/api/checks/SecondaryLocation.java

```diff
@@ -20,31 +20,35 @@
 package org.sonar.iac.common.api.checks;
 
 import java.util.Objects;
+import javax.annotation.Nullable;
 import org.sonar.iac.common.api.tree.HasTextRange;
 import org.sonar.iac.common.api.tree.impl.TextRange;
-import org.sonar.iac.common.api.tree.impl.TextRanges;
 
 public class SecondaryLocation {
 
   public final TextRange textRange;
 
   public final String message;
 
+  @Nullable
+  public final String filePath;
```

Comment:
Maybe add a description, that `filePath` doesn't have to be strictly absolute or relative, but has to satisfy `FilePredicates.is`, and that if it's `null`, then the issue will be raised on the same file?